drivers/mrf24j40/mrf24j40_internal.c

```diff
@@ -36,6 +36,56 @@ static inline void getbus(mrf24j40_t *dev)
     spi_acquire(SPIDEV, CSPIN, SPI_MODE_0, dev->params.spi_clk);
 }
 
+void mrf24j40_enable_pa_lna(mrf24j40_t *dev)
+{
+#if MRF24J40_USE_EXT_PA_LNA
+    /* Configure enable pin of the Voltage Regulator for the PA (GPIO3) on MRF24J40MC */
+    mrf24j40_reg_write_short(dev, MRF24J40_REG_TRISGPIO, 1 << 3);
+
+    /* Enable the volate regulator to power the Power Amplifier */
+    mrf24j40_reg_write_short(dev, MRF24J40_REG_GPIO, 1 << 3);
+
+    mrf24j40_reg_write_long(dev, MRF24J40_REG_TESTMODE, (MRF24J40_TESTMODE_RSSIWAIT0 |
+                                                         MRF24J40_TESTMODE_TESTMODE2 |
+                                                         MRF24J40_TESTMODE_TESTMODE1 |
+                                                         MRF24J40_TESTMODE_TESTMODE0));
+#else
+    (void) dev;
+#endif
+}
+
+void mrf24j40_disable_pa_lna(mrf24j40_t *dev)
+{
+#if MRF24J40_USE_EXT_PA_LNA
+    /* Disable automatic switch on PA/LNA */
+    mrf24j40_reg_write_long(dev, MRF24J40_REG_TESTMODE, MRF24J40_TESTMODE_RSSIWAIT0);
+
+    /* Configure all GPIOs as Output */
+    mrf24j40_reg_write_short(dev, MRF24J40_REG_TRISGPIO, 0xF);
+
+    /* Disable all GPIO outputs */
+    mrf24j40_reg_write_short(dev, MRF24J40_REG_GPIO, 0);
+#else
+    (void) dev;
+#endif
+}
+
+void mrf24j40_enable_lna(mrf24j40_t *dev)
+{
+#if MRF24J40_USE_EXT_PA_LNA
+    /* Disable automatic switch on PA/LNA */
+    mrf24j40_reg_write_long(dev, MRF24J40_REG_TESTMODE, MRF24J40_TESTMODE_RSSIWAIT0);
+
+    /* Configure all GPIOs as Output */
+    mrf24j40_reg_write_short(dev, MRF24J40_REG_TRISGPIO, 0xF);
+
+    /* Enable LNA */
+    mrf24j40_reg_write_short(dev, MRF24J40_REG_GPIO, 0xC);
```

Comment:
For `lna` you need bit2 and bit3 whereas `pa_lna` only required bit3?

Reply:
When the three TESTMODE bits in the TESTMODE register are set, mrf24j40 will controls those pins autonomously. (see [Section 1.3.1 PA/LNA CONTROL](http://ww1.microchip.com/downloads/en/devicedoc/75002a.pdf)) That single pin/bit enables the voltage regulator for the PA thats only present on the C version of the module and that is not controlled by the mrf24j40 automatically.

Reply:
When you write `0xC` (`0b00001100`) to the register, it sets GPIO1 to zero, and GPIO2/GPIO3 to one. 1.3.1 says:

"*Set GPIO2 (0x33<2>) = 1 and GPIO1 (0x32<1>) = 0. This enables the LNA and disables the PA"*

So why do you set GPIO3 here? I think it would be clearer to use preprocessor constants

Reply:
GPIO3 is connected to the PA voltage regulator, so would indeed not need to be enabled here.
So I guess writing `0x4` should suffice here. I just copied that constant from the Microchip MiWi SDK driver - maybe the voltage regulator has a startup time that has to be considered, so shutting it off for short periods would be detrimental.

Reply:
Why don't you just read, modify the bits you need to modify, and write back? That's what I intended to say above as well. Furthermore, I needed to understand the relation between your code and the documentation.

Reply:
I thought when the driver from Microchip doesn't bother with it, I could get away without doing so too 😉 

But I've added `mrf24j40_reg_and_short` / `mrf24j40_reg_or_short` now and use explicit defines for the GPIOs.